File: UDPClient.java
package udp;

import java.io.*;
import java.net.*;
import java.util.Scanner;

/**
 * UDP Client with interactive interface
 * Supports multiple functionalities from the project requirements
 */
class UDPClient {
    private static final String SERVER_HOST = "localhost";
    private static final int SERVER_PORT = 9876;
    private static final int BUFFER_SIZE = 1024;
    
    public static void main(String[] args) throws Exception {
        Scanner scanner = new Scanner(System.in);
        DatagramSocket clientSocket = new DatagramSocket();
        InetAddress serverAddress = InetAddress.getByName(SERVER_HOST);
        
        System.out.println("UDP Client ishga tushdi!");
        System.out.println("Server: " + SERVER_HOST + ":" + SERVER_PORT);
        System.out.println();
        System.out.println("Mavjud buyruqlar:");
        System.out.println("- 'salom' - Server bilan salomlashish");
        System.out.println("- 'status' - Server holatini tekshirish");
        System.out.println("- 'vaqt' - Hozirgi vaqtni olish");
        System.out.println("- 'tasodifiy son' - Tasodifiy son olish");
        System.out.println("- 'haroratni yuboring' - Sensor ma'lumoti");
        System.out.println("- 'menga maqol ayting' - Maqol olish");
        System.out.println("- 'ip manzilim qanday?' - IP manzilni olish");
        System.out.println("- 'foydalanuvchilar' - Faol foydalanuvchilar");
        System.out.println("- 'fayl:filename.txt' - Fayl mavjudligini tekshirish");
        System.out.println("- 'id:1' - Ma'lumotlar bazasidan ma'lumot");
        System.out.println("- 'tosh/qaychi/qog'oz' - O'yin");
        System.out.println("- '5 + 3' - Matematik amal");
        System.out.println("- 'json' - JSON formatda ma'lumot");
        System.out.println("- 'exit' - Dasturdan chiqish");
        System.out.println();
        
        while (true) {
            try {
                System.out.print("Xabar yuboring: ");
                String userInput = scanner.nextLine();
                
                if (userInput.equalsIgnoreCase("exit")) {
                    System.out.println("Dasturdan chiqilmoqda...");
                    break;
                }
                
                if (userInput.trim().isEmpty()) {
                    System.out.println("Bo'sh xabar yuborilmaydi!");
                    continue;
                }
                
                // Send message to server
                byte[] sendData = userInput.getBytes();
                DatagramPacket sendPacket = new DatagramPacket(
                    sendData, sendData.length, serverAddress, SERVER_PORT
                );
                clientSocket.send(sendPacket);
                
                // Receive response from server
                byte[] receiveData = new byte[BUFFER_SIZE];
                DatagramPacket receivePacket = new DatagramPacket(
                    receiveData, receiveData.length
                );
                clientSocket.receive(receivePacket);
                
                String serverResponse = new String(receivePacket.getData()).trim();
                System.out.println("SERVERDAN: " + serverResponse);
                System.out.println();
                
            } catch (Exception e) {
                System.err.println("Xatolik: " + e.getMessage());
                System.out.println("Qayta urinib ko'ring...");
            }
        }
        
        clientSocket.close();
        scanner.close();
    }
}

File: udp_client.py
#!/usr/bin/env python3
"""
UDP Client with interactive interface
Supports multiple functionalities from the project requirements
"""

import socket
import sys

class UDPClient:
    def __init__(self, host='localhost', port=9876):
        self.host = host
        self.port = port
        self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
        
    def start(self):
        """Start the UDP client"""
        print("UDP Client ishga tushdi!")
        print(f"Server: {self.host}:{self.port}")
        print()
        print("Mavjud buyruqlar:")
        print("- 'salom' - Server bilan salomlashish")
        print("- 'status' - Server holatini tekshirish")
        print("- 'vaqt' - Hozirgi vaqtni olish")
        print("- 'tasodifiy son' - Tasodifiy son olish")
        print("- 'haroratni yuboring' - Sensor ma'lumoti")
        print("- 'menga maqol ayting' - Maqol olish")
        print("- 'ip manzilim qanday?' - IP manzilni olish")
        print("- 'foydalanuvchilar' - Faol foydalanuvchilar")
        print("- 'fayl:filename.txt' - Fayl mavjudligini tekshirish")
        print("- 'id:1' - Ma'lumotlar bazasidan ma'lumot")
        print("- 'tosh/qaychi/qog'oz' - O'yin")
        print("- '5 + 3' - Matematik amal")
        print("- 'json' - JSON formatda ma'lumot")
        print("- 'exit' - Dasturdan chiqish")
        print()
        
        while True:
            try:
                user_input = input("Xabar yuboring: ").strip()
                
                if user_input.lower() == 'exit':
                    print("Dasturdan chiqilmoqda...")
                    break
                
                if not user_input:
                    print("Bo'sh xabar yuborilmaydi!")
                    continue
                
                # Send message to server
                self.socket.sendto(user_input.encode('utf-8'), (self.host, self.port))
                
                # Receive response from server
                data, addr = self.socket.recvfrom(1024)
                server_response = data.decode('utf-8')
                
                print(f"SERVERDAN: {server_response}")
                print()
                
            except KeyboardInterrupt:
                print("\nDasturdan chiqilmoqda...")
                break
            except Exception as e:
                print(f"Xatolik: {e}")
                print("Qayta urinib ko'ring...")
        
        self.socket.close()

def main():
    """Main function to start the client"""
    client = UDPClient()
    try:
        client.start()
    except KeyboardInterrupt:
        print("\nClient to'xtatilmoqda...")

if __name__ == "__main__":
    main()
